#Methods
***

## Primary analysis methods

Members of the consortium have developed pipelines to analyse the data produced by the project consistently.  

 - [BS-seq](#/md/bs_seq) reads are aligned using GEM, and methylation status is deterimned with bs_call. 
 - [ChIP-seq](#/md/chip_seq) reads are aligned using BWA and enriched regions are called using MACS2.
 - [DNAse-seq](#/md/dnase_seq) reads are aligned using BWA and enriched regions are called using MACS2.
 - [RNA-seq](#/md/rna_seq) reads are aligned using GEM and transcription is quantified with FLUX capacitor.
 - For a subset of samples (progenitor cells), [RNA-seq](#/md/rna_seq_cu) reads are aligned to the transcriptome using bowtie and transcription is quantified with MMSEQ.
 
<!-- ##Secondary analysis methods-->
  
## Experimental methods
 - Experimental protocols are documented on the main [BLUEPRINT website](http://www.blueprint-epigenome.eu/index.cfm?p=7BF8A4B6-F4FE-861A-2AD57A08D63D0B58)


 